src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -40,6 +41,9 @@ class spark_resource_adaptor final : public rmm::mr::device_memory_resource {
     if (env->GetJavaVM(&jvm) < 0) {
       throw std::runtime_error("GetJavaVM failed");
     }
+    retry_oom_class = env->FindClass("com/nvidia/spark/rapids/jni/RetryOOM");
+    split_and_retry_oom_class = env->FindClass("com/nvidia/spark/rapids/jni/SplitAndRetryOOM");
+    cudf_exception_class = env->FindClass("ai/rapids/cudf/CudfException");
```

Comment:
This risks the JVM garbage collecting these classes unless we create an explicit reference (either directly or indirectly).

Reply:
Yeah I think this is related to what I am seeing.

Reply:
also should be fixed